simulator/src/circuit.js

```
@@ -53,93 +54,101 @@
-            if (dependencies === '') {
-                dependencies = scopeList[id].name;
-            } else {
-                dependencies += `, ${scopeList[id].name}`;
+    try {
+        const scope = scopeList[scopeId];
+        if (Object.keys(scopeList).length <= 1) {
+            showError('At least 2 circuits need to be there in order to delete a circuit.');
+            return;
+        }
+        let dependencies = '';
+        for (id in scopeList) {
```

Biome

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)